crates/ruff_cli/src/diagnostics.rs

```diff
@@ -85,7 +92,39 @@ pub fn lint_path(
     debug!("Checking: {}", path.display());
 
     // Read the file from disk.
-    let contents = std::fs::read_to_string(path)?;
+    let (contents, jupyter_index) = if is_jupyter_notebook(path) {
+        // See also the black implementation
+        // <https://github.com/psf/black/blob/69ca0a4c7a365c5f5eea519a90980bab72cab764/src/black/__init__.py#L1017-L1046>
+        let notebook = match read_jupyter_notebook(path) {
+            Ok(Some(notebook)) => notebook,
+            Ok(None) => {
+                // Not a python notebook, this could e.g. be an R notebook which we want to just skip
+                return Ok(Diagnostics {
+                    messages: vec![],
+                    fixed: FxHashMap::default(),
+                    jupyter_index: FxHashMap::default(),
+                });
+            }
+            Err(diagnostic) => {
+                // Failed to read the jupyter notebook
+                return Ok(Diagnostics {
+                    messages: vec![Message::from_diagnostic(
+                        *diagnostic,
+                        path.to_string_lossy().to_string(),
+                        None,
+                        1,
+                    )],
+                    fixed: FxHashMap::default(),
+                    jupyter_index: FxHashMap::default(),
+                });
```

Comment:
You can simplify this if `Diagnostics` implements `Default`
```suggestion
                return Ok(Diagnostics {
                	..Diagnostics::default(),
                	messages: vec![Message::from_diagnostic(
                        *diagnostic,
                        path.to_string_lossy().to_string(),
                        None,
                        1,
                    )],
                });
```